solidity/contracts/hooks/OPL2ToL1CcipReadHook.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {Message} from "../libs/Message.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+import {IMailbox} from "../interfaces/IMailbox.sol";
+import {IPostDispatchHook} from "../interfaces/hooks/IPostDispatchHook.sol";
+import {StandardHookMetadata} from "./libs/StandardHookMetadata.sol";
+import {AbstractPostDispatchHook} from "./libs/AbstractPostDispatchHook.sol";
+import {AbstractMessageIdAuthHook} from "./libs/AbstractMessageIdAuthHook.sol";
+import {InterchainGasPaymaster} from "../hooks/igp/InterchainGasPaymaster.sol";
+
+/**
+ * @title OPL2ToL1CcipReadHook
+ * @notice Inform an OPL2ToL1ProveWithdrawalIsm that a withdrawal has been initiated on L2
+ * @dev We expect a single CCIP-read ISM executing portal.proveWithdrawal() and portal.finalizeWithdrawal() on L1 after 7 days. This is due to the fact that OP Stack expect the proof submitter and the finalizer to be the same caller.
+ */
+contract OPL2ToL1CcipReadHook is AbstractPostDispatchHook {
```

Comment:
So I haven't been able to think too much about it yet, but have you given it some thought how to share/merge logic with the existing OpStackL2toL1 hook and isms which are used for securing a message (while these ones are primarily concerned with moving value), curious on your assessment. I know @yorhodes would like to ideally see it unified, but admittedly we haven't really given much guidance there

Reply:
> What do you think?
I think it depends on the use-case, the former fits better when the devs want to secure messages depending on some runtime conditions/logic. While the latter it would be a better fit when you always want the messages delivered securely, regardless

> Did you come up with that yourself or chatted with @yorhodes about?
That's my personal brainstorming, happy to hear @yorhodes opinion too